BlazarUI/app/scripts/components/sidebar/SidebarItem.jsx

```diff
@@ -1,92 +1,161 @@
+      return (<span />);
+    }
 
     return (
-      <div className='sidebar-item__branch-link'>
-        <Icon type='octicon' name='git-branch' classNames='icon-muted'/>{ ' ' }
-        <span className='sidebar-item__module-branch-name'>
-          {truncate(gitInfo.branch, 40, true)}
-        </span>
-      </div>
-    )
-  }
-  
-  renderBuildLink() {
-    const {build, prevBuildState} = this.props;
-    let icon, buildIdLink;
-    
-    if (prevBuildState) {
-      icon = (
-        <Link to={build.blazarPath} className='sidebar-item__building-icon-link'>
-          {buildResultIcon(build.state, prevBuildState)}
+      <span className='sidebar-item__module-branch-name'>
+        <Icon type='octicon' name='git-branch' classNames='repo-octicon'/>{ '   ' }
+        <Link to={lastBuild.blazarPath}>
+          {truncate(gitInfo.branch, 20, true)}
         </Link>
-      );
+      </span>
+    );
+  }
 
-      buildIdLink =(
-        <Link to={build.blazarPath} className='sidebar-item__build-number'>
-          #{build.buildNumber}
-        </Link>
-      );  
+  renderBuildIcon(buildState) {
+    if (buildState === BuildStates.SUCCEEDED) {
+      return (<div />);
     }
 
-    // never built on blazar
-    else {
-      icon = (
-        <div className='sidebar-item__building-icon-link'>
-          {buildResultIcon(build.state, 'never-built')}
+    return (
+      <div className='sidebar-item__building-icon-link'>
+        {buildResultIcon(buildState)}
+      </div>
+    );
+  }
+
+  renderBranchRow(build) {
+    const gitInfo = build.gitInfo;
+    const lastBuild = build.lastBuild;
+
+    if (lastBuild === undefined) {
+      return (
+        <div>
+          {this.renderBranchText(build)}
         </div>
       );
     }
-    
+
+    let buildState = build.inProgressBuild !== undefined ? build.inProgressBuild.state : lastBuild.state;
+
     return (
-      <div>
-        {icon}
-        {buildIdLink}
+      <div className='sidebar-item__branch-link'>
+        {this.renderBuildIcon(buildState)}
+        {this.renderBranchText(build)}
+        <Link to={lastBuild.blazarPath} className='sidebar-item__build-number'>
+          #{lastBuild.buildNumber}
+        </Link>
       </div>
     );
   }
 
-  render() {    
-    const {build, gitInfo} = this.props;
+  renderBranchRows() {
+    const {builds} = this.props;
+    let realBuilds = builds.slice();
+
+    if (realBuilds.length > 3 && !this.state.expanded) {
+      const originalSize = realBuilds.length;
+      const splicedBuilds = realBuilds.splice(0, 3);
+      const numberRemaining = originalSize - 3;
+
+      return (
+        <div>
+          {splicedBuilds.map((build) => {return this.renderBranchRow(build);})}       
+          {this.renderExpandText(numberRemaining)}
+        </div>
+      );
+    } 
+
+    else {
+      return (
+        <div>
+          {realBuilds.map((build) => {return this.renderBranchRow(build);})}
+          {this.renderExpandText(0)}
```

Comment:
not a huge deal but you can use an ES6 default function parameter in `renderExpandText` so you don't need to pass 0 here.